### PR TITLE
[Feature] CheckboxGroup component

### PR DESCRIPTION
**Pull request by ketsappi**
## Description

- Add CheckboxGroup component
- Update Checkbox label font size to be smaller

## Motivation and Context

 Add CheckboxGroup component for visually helping grouping checkboxes together. Checkbox font size was decreased so they work visually better when grouped; CheckboxGroup and Checkbox has all the font sizes as 16px.

## How Has This Been Tested?

Locally in styleguidist and CRA-TS test project.

## Screenshots (if appropriate):

Example of the CheckboxGroup

## Release notes

- Add Checkboxgroup component.
- Update Checkbox label's font size to be smaller

### CODE REVIEW

File: src/core/Form/Checkbox/CheckboxGroup.baseStyles.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { element, font } from '../../theme/reset';
+
+export const baseStyles = (theme: SuomifiTheme) => css`
+  ${element(theme)}
+  ${font(theme)('bodyText')}
+
+  & .fi-checkbox_container {
+    margin-top: ${theme.spacing.xs};
+  }
+
```
Comment (LJKaski) [inline src/core/Form/Checkbox/CheckboxGroup.baseStyles.tsx:12]: In the designs there's a 10px margin between the label and help text. Should it be added here as well or was something else decided on it?

File: src/core/Form/Checkbox/Checkbox.md [hunk 2]
```diff
@@ -52,3 +52,29 @@ const [checked, setChecked] = React.useState(false);
...
+    <Checkbox defaultChecked hintText="Example hint text">
+      Regular checkbox that is checked and has a hint text
+    </Checkbox>
+
+    <Checkbox hintText="Example hint text">
+      Regular checkbox with a hint text
+    </Checkbox>
+
+    <Checkbox hintText="Example hint text">
+      Regular checkbox that has a hint text
```
Comment (LJKaski) [inline src/core/Form/Checkbox/Checkbox.md:76]: I'm slightly bothered that this is identical with the previous one bat the label/content is different 😅 could the last one be without hint text to show that's possible as well?

File: src/core/Form/Checkbox/CheckboxGroup.baseStyles.tsx (hunk 1 above)
Comment (ketsappi) [reply src/core/Form/Checkbox/CheckboxGroup.baseStyles.tsx:12]: Not intentionally denied to left out. Was creating this based on the RadiobuttonGroup and went the same road for not telling the explicit margin between label and hint text.

I will add the margin.

File: src/core/Form/Checkbox/Checkbox.md (hunk 2 above)
Comment (ketsappi) [reply src/core/Form/Checkbox/Checkbox.md]: The latter example is now changed to be just "_Regular checkbox_"

File: src/core/Form/Checkbox/CheckboxGroup.baseStyles.tsx (hunk 1 above)
Comment (ketsappi) [reply src/core/Form/Checkbox/CheckboxGroup.baseStyles.tsx:11]: Margin added.

The logic for showing the margin between label, hint text and between actual group was added.
Also the large variant in Checkboxes have different margin, so therefore it uses also top margin to take care of that.

**Review by LJKaski — APPROVED** [review]
Looks good and seems to be working well. Good job! 👍

**Pull request merged**